### PR TITLE
Migrate deferred_components_prebuild_validator to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart [hunk 1]
```diff
@@ -70,16 +67,20 @@ class DeferredComponentsPrebuildValidator extends DeferredComponentsValidator {
...
         // generate into temp directory
         final Map<String, List<File>> results =
-          await androidFiles.generateFiles(
-            alternateAndroidDir: outputDir,
-            clearAlternateOutputDir: true,
-          );
-        for (final File file in results['outputs']) {
-          generatedFiles.add(file.path);
-          changesMade = true;
+        await androidFiles.generateFiles(
```
Comment (GaryQian) [inline packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart:70]: This should be indented one more layer!

**Review by GaryQian — APPROVED** [review]
Minor formatting nit, otherwise LGTM!

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/android/deferred_components_prebuild_validator.dart]: Formatted.

**Pull request merged**